biorxiv/corpora_comparison/biorxiv_pmc_corpora_comparison.py

```diff
@@ -0,0 +1,445 @@
+        "doi":"last",
+        "published_doi":"last",
+        "journal":"last",
+        "pmcid":"last"
+    })
+    .reset_index(drop=True)
+)
+mapped_doi_df.tail()
+
+
+# In[18]:
+
+
+print(f"Total # of Preprints Mapped: {mapped_doi_df.shape[0]}")
+print(f"Total % of Mapped: {mapped_doi_df.shape[0]/71118}")
+
+
+# In[19]:
+
+
+preprint_count = aggregate_word_counts([
+    Path(f"output/biorxiv_word_counts/{Path(file).stem}.tsv")
+    for file in mapped_doi_df.document.values.tolist()
+    if Path(f"output/biorxiv_word_counts/{Path(file).stem}.tsv").exists()
+])
+
+
+# In[20]:
+
+
+published_count = aggregate_word_counts([
+    Path(f"output/pmc_word_counts/{file}.tsv")
+    for file in mapped_doi_df.pmcid.values.tolist()
+    if Path(f"output/pmc_word_counts/{file}.tsv").exists()
+])
+
+
+# In[21]:
+
+
+preprint_count, published_count = remove_stop_words(
+    preprint_count,
+    published_count
+)
+
+
+# In[22]:
+
+
+top_hundred_preprint = preprint_count.most_common(100)
+top_hundred_preprint[0:10]
+
+
+# In[23]:
+
+
+top_hundred_published = published_count.most_common(100)
+top_hundred_published[0:10]
+
+
+# In[24]:
+
+
+print("Number of words in preprint but not in published version:")
+preprint_difference = set(list(preprint_count.keys())) - set(list(published_count.keys()))
+print(len(preprint_difference))
+
+
+# In[25]:
+
+
+[
+    preprint_difference.pop()
+    for i in range(10)
+]
+
+
+# In[26]:
+
+
+print("Number of words in published version but not in preprint:")
+published_difference = set(list(published_count.keys())) - set(list(preprint_count.keys()))
+print(len(published_difference))
+
+
+# In[27]:
+
+
+[
+    published_difference.pop()
+    for i in range(10)
+]
+
+
+# In[28]:
+
+
+total_words = set(list(dict(top_hundred_preprint).keys()) + list(dict(top_hundred_published).keys()))
+data = []
+for word in tqdm_notebook(total_words):
+    
+    word_stat = get_term_statistics(
+        preprint_count,
+        published_count,
+        word
+    )
+    
+    data.append({
+        "lemma": word,
+        "preprint_count":preprint_count[word] if word in preprint_count else 0,
+        "published_count":published_count[word] if word in published_count else 0,
+        "preprint_total":word_stat['chi_sq'][3][1,0],
+        "published_total":word_stat['chi_sq'][3][1,1],
+        "log_p": word_stat['chi_sq'][1],
+        "log_likelihood": word_stat['log_likelihood'],
+        "odds_ratio": word_stat['odds_ratio']
+    })
+
+
+# In[29]:
+
+
+published_comparison_stats_df = pd.DataFrame.from_records(data)
+published_comparison_stats_df.to_csv(
+    "output/preprint_to_published_comparison.tsv", 
+    sep="\t", index=False
+)
+published_comparison_stats_df.head()
+
+
+# In[30]:
+
+
+(
+    published_comparison_stats_df
+    .sort_values("log_likelihood", ascending=False)
+    .head(20)
+)
+
+
+# In[31]:
+
+
+(
+    published_comparison_stats_df
+    .sort_values("odds_ratio", ascending=False)
+    .head(20)
+)
+
```

Comment:
Very cool! Do you have any takeaways from these tables? Are the words that are more common in published manuscripts added due to peer review? Or editing? Or changes in style between the draft and the final version?

You may want to add a few sentences of discussion to the bottom of the notebook with the big points of the two analyses